### PR TITLE
[PDI-13271]: External Ids are not handled properly when incoming key is null

### PR DESCRIPTION
**Pull request by TatsianaKasiankova**
The problem was:
when isUseExtId=Y for the field, we add incorrect field name in FieldsToNull.
What was done:
1)Added utility method to get valid names for null fields candidates.
2)Added calls this method in the steps: SalesforceUpsert, SalesforceInput, SalesforceUpdate.
3)Added a couple of unit tests.

### CODE REVIEW

**Comment by TatsianaKasiankova** [comment]
@rpbouman , @pamval Hi, please review the fix.

**Comment by TatsianaKasiankova** [comment]
@rpbouman Hi, please review the fix.

**Comment by brosander** [comment]
Changed files:
==============
  plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceinsert/SalesforceInsert.java
  plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceupdate/SalesforceUpdate.java
  plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceupsert/SalesforceUpsert.java
  plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceutils/SalesforceUtils.java
  plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceutils/messages/messages_en_US.properties
  plugins/salesforce/test-src/org/pentaho/di/trans/steps/salesforceutils/SalesforceUtilsTest.java

Checkstyle violations:
----------------------
No errors found




Unit test coverage change:
============
org.pentaho.di.trans.steps.salesforceinsert.SalesforceInsert:
 * instructionPercentCoverageChange :**-0.21%**
    
org.pentaho.di.trans.steps.salesforceutils.SalesforceUtils:
 * branchPercentCoverage :58.33%
 * linePercentCoverage :68.42%
 * methodPercentCoverage :66.67%
 * complexityPercentCoverage :44.44%
 * instructionPercentCoverage :57.94%

**Comment by TatsianaKasiankova** [comment]
@rpbouman, @brosander Please review this PR.

**Comment by lucboudreau** [comment]
Build Commands:
==============
Build:
------
1. rm -r ~/.kettle; rm -r ~/.pentaho; echo removed hidden folders
2. ant -Djunit.jvmargs=-XX:PermSize=256m -f plugins/salesforce/build.xml clean-all resolve jacoco jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceinsert/SalesforceInsert.java
  plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceupdate/SalesforceUpdate.java
  plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceupsert/SalesforceUpsert.java
  plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceutils/SalesforceUtils.java
  plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceutils/messages/messages_en_US.properties
  plugins/salesforce/test-src/org/pentaho/di/trans/steps/salesforceutils/SalesforceUtilsTest.java


Checkstyle violations:
======================
See: https://github.com/pentaho/pentaho-coding-standards

File plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceupsert/SalesforceUpsert.java
1. Message: '(' is not followed by whitespace. at Line: 136 Column: 30









Unit test coverage change:
============
org.pentaho.di.trans.steps.salesforceinsert.SalesforceInsert:
 * instructionPercentCoverageChange :**-0.21%**
    
org.pentaho.di.trans.steps.salesforceutils.SalesforceUtils:
 * branchPercentCoverage :58.33%
 * linePercentCoverage :68.42%
 * methodPercentCoverage :66.67%
 * complexityPercentCoverage :44.44%
 * instructionPercentCoverage :57.94%

**Comment by brosander** [comment]
@TatsianaKasiankova please fix checkstyle, also we're going for 80% coverage on new code so please increase coverage on SalesforceUtils and cover your change to SalesforceInsert.

**Comment by TatsianaKasiankova** [comment]
@brosander Hi, Bryan
I've added unit tests and fixed checkstyle violations.
Please review it.

**Comment by lucboudreau** [comment]
Build Commands:
==============
Build:
------
1. rm -r ~/.kettle; rm -r ~/.pentaho; echo removed hidden folders
2. ant -Djunit.jvmargs=-XX:PermSize=256m -f plugins/salesforce/build.xml clean-all resolve jacoco jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceinsert/SalesforceInsert.java
  plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceupdate/SalesforceUpdate.java
  plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceupsert/SalesforceUpsert.java
  plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceutils/SalesforceUtils.java
  plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceutils/messages/messages_en_US.properties
  plugins/salesforce/test-src/org/pentaho/di/trans/steps/salesforceinsert/SalesforceInsertTest.java
  plugins/salesforce/test-src/org/pentaho/di/trans/steps/salesforceupdate/SalesforceUpdateTest.java
  plugins/salesforce/test-src/org/pentaho/di/trans/steps/salesforceupsert/SalesforceUpsertTest.java
  plugins/salesforce/test-src/org/pentaho/di/trans/steps/salesforceutils/SalesforceUtilsTest.java


Checkstyle violations:
======================
See: https://github.com/pentaho/pentaho-coding-standards

File plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceinsert/SalesforceInsert.java
1. Message: '<' is not followed by whitespace. at Line: 132 Column: 19
2. Message: '<' is not followed by whitespace. at Line: 132 Column: 64
3. Message: '<' is not preceded with whitespace. at Line: 132 Column: 18
4. Message: '<' is not preceded with whitespace. at Line: 132 Column: 63
5. Message: '>' is not followed by whitespace. at Line: 132 Column: 79
6. Message: '>' is not preceded with whitespace. at Line: 132 Column: 33
7. Message: '>' is not preceded with whitespace. at Line: 132 Column: 78
8. Message: '<' is not followed by whitespace. at Line: 134 Column: 19
9. Message: '<' is not followed by whitespace. at Line: 134 Column: 56
10. Message: '<' is not preceded with whitespace. at Line: 134 Column: 18
11. Message: '<' is not preceded with whitespace. at Line: 134 Column: 55
12. Message: '>' is not followed by whitespace. at Line: 134 Column: 63
13. Message: '>' is not preceded with whitespace. at Line: 134 Column: 25
14. Message: '>' is not preceded with whitespace. at Line: 134 Column: 62
15. Message: '<' is not followed by whitespace. at Line: 57 Column: 24
16. Message: '<' is not preceded with whitespace. at Line: 57 Column: 23
17. Message: '>' is not preceded with whitespace. at Line: 57 Column: 25
18. Message: '?' is not followed by whitespace. at Line: 57 Column: 25
19. Message: '?' is not preceded with whitespace. at Line: 57 Column: 24

File plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceupdate/SalesforceUpdate.java
1. Message: '<' is not followed by whitespace. at Line: 129 Column: 19
2. Message: '<' is not followed by whitespace. at Line: 129 Column: 56
3. Message: '<' is not preceded with whitespace. at Line: 129 Column: 18
4. Message: '<' is not preceded with whitespace. at Line: 129 Column: 55
5. Message: '>' is not followed by whitespace. at Line: 129 Column: 63
6. Message: '>' is not preceded with whitespace. at Line: 129 Column: 25
7. Message: '>' is not preceded with whitespace. at Line: 129 Column: 62
8. Message: '<' is not followed by whitespace. at Line: 130 Column: 19
9. Message: '<' is not followed by whitespace. at Line: 130 Column: 64
10. Message: '<' is not preceded with whitespace. at Line: 130 Column: 18
11. Message: '<' is not preceded with whitespace. at Line: 130 Column: 63
12. Message: '>' is not followed by whitespace. at Line: 130 Column: 79
13. Message: '>' is not preceded with whitespace. at Line: 130 Column: 33
14. Message: '>' is not preceded with whitespace. at Line: 130 Column: 78
15. Message: '<' is not followed by whitespace. at Line: 54 Column: 24
16. Message: '<' is not preceded with whitespace. at Line: 54 Column: 23
17. Message: '>' is not preceded with whitespace. at Line: 54 Column: 25
18. Message: '?' is not followed by whitespace. at Line: 54 Column: 25
19. Message: '?' is not preceded with whitespace. at Line: 54 Column: 24

File plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceupsert/SalesforceUpsert.java
1. Message: '<' is not followed by whitespace. at Line: 127 Column: 19
2. Message: '<' is not followed by whitespace. at Line: 127 Column: 56
3. Message: '<' is not preceded with whitespace. at Line: 127 Column: 18
4. Message: '<' is not preceded with whitespace. at Line: 127 Column: 55
5. Message: '>' is not followed by whitespace. at Line: 127 Column: 63
6. Message: '>' is not preceded with whitespace. at Line: 127 Column: 25
7. Message: '>' is not preceded with whitespace. at Line: 127 Column: 62
8. Message: '<' is not followed by whitespace. at Line: 128 Column: 19
9. Message: '<' is not followed by whitespace. at Line: 128 Column: 64
10. Message: '<' is not preceded with whitespace. at Line: 128 Column: 18
11. Message: '<' is not preceded with whitespace. at Line: 128 Column: 63
12. Message: '>' is not followed by whitespace. at Line: 128 Column: 79
13. Message: '>' is not preceded with whitespace. at Line: 128 Column: 33
14. Message: '>' is not preceded with whitespace. at Line: 128 Column: 78
15. Message: '<' is not followed by whitespace. at Line: 53 Column: 24
16. Message: '<' is not preceded with whitespace. at Line: 53 Column: 23
17. Message: '>' is not preceded with whitespace. at Line: 53 Column: 25
18. Message: '?' is not followed by whitespace. at Line: 53 Column: 25
19. Message: '?' is not preceded with whitespace. at Line: 53 Column: 24

File plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceutils/SalesforceUtils.java
1. Message: '<' is not followed by whitespace. at Line: 38 Column: 24
2. Message: '<' is not preceded with whitespace. at Line: 38 Column: 23
3. Message: '>' is not preceded with whitespace. at Line: 38 Column: 25
4. Message: '?' is not followed by whitespace. at Line: 38 Column: 25
5. Message: '?' is not preceded with whitespace. at Line: 38 Column: 24

File plugins/salesforce/test-src/org/pentaho/di/trans/steps/salesforceinsert/SalesforceInsertTest.java
1. Message: '<' is not followed by whitespace. at Line: 29 Column: 26
2. Message: '<' is not preceded with whitespace. at Line: 29 Column: 25
3. Message: '>' is not preceded with whitespace. at Line: 29 Column: 68
4. Message: '<' is not followed by whitespace. at Line: 34 Column: 28
5. Message: '<' is not preceded with whitespace. at Line: 34 Column: 27
6. Message: '>' is not followed by whitespace. at Line: 34 Column: 71
7. Message: '>' is not preceded with whitespace. at Line: 34 Column: 70

File plugins/salesforce/test-src/org/pentaho/di/trans/steps/salesforceupdate/SalesforceUpdateTest.java
1. Message: '<' is not followed by whitespace. at Line: 29 Column: 26
2. Message: '<' is not preceded with whitespace. at Line: 29 Column: 25
3. Message: '>' is not preceded with whitespace. at Line: 29 Column: 68
4. Message: '<' is not followed by whitespace. at Line: 34 Column: 28
5. Message: '<' is not preceded with whitespace. at Line: 34 Column: 27
6. Message: '>' is not followed by whitespace. at Line: 34 Column: 71
7. Message: '>' is not preceded with whitespace. at Line: 34 Column: 70

File plugins/salesforce/test-src/org/pentaho/di/trans/steps/salesforceupsert/SalesforceUpsertTest.java
1. Message: '<' is not followed by whitespace. at Line: 29 Column: 26
2. Message: '<' is not preceded with whitespace. at Line: 29 Column: 25
3. Message: '>' is not preceded with whitespace. at Line: 29 Column: 68
4. Message: '<' is not followed by whitespace. at Line: 34 Column: 28
5. Message: '<' is not preceded with whitespace. at Line: 34 Column: 27
6. Message: '>' is not followed by whitespace. at Line: 34 Column: 71
7. Message: '>' is not preceded with whitespace. at Line: 34 Column: 70


License header violations:
==========================

1. plugins/salesforce/test-src/org/pentaho/di/trans/steps/salesforceinsert/SalesforceInsertTest.java

2. plugins/salesforce/test-src/org/pentaho/di/trans/steps/salesforceupdate/SalesforceUpdateTest.java

3. plugins/salesforce/test-src/org/pentaho/di/trans/steps/salesforceupsert/SalesforceUpsertTest.java








Unit test coverage change:
============
org.pentaho.di.trans.steps.salesforceupdate.SalesforceUpdateMeta:
 * linePercentCoverage :0.58%
 * methodPercentCoverage :2.63%
 * complexityPercentCoverage :1.85%
 * instructionPercentCoverage :0.31%
    
org.pentaho.di.trans.steps.salesforceupsert.SalesforceUpsertMeta:
 * linePercentCoverage :0.52%
 * methodPercentCoverage :2.38%
 * complexityPercentCoverage :1.69%
 * instructionPercentCoverage :0.28%
    
org.pentaho.di.trans.steps.salesforceinsert.SalesforceInsert:
 * instructionPercentCoverageChange : +5.13%
 * branchPercentCoverageChange : +7.50%
 * linePercentCoverageChange : +2.05%
 * complexityPercentCoverageChange : +10.64%
    
org.pentaho.di.trans.steps.salesforceupdate.SalesforceUpdate:
 * branchPercentCoverage :19.44%
 * linePercentCoverage :24.43%
 * methodPercentCoverage :57.14%
 * complexityPercentCoverage :20.93%
 * instructionPercentCoverage :24.47%
    
org.pentaho.di.trans.steps.salesforceupsert.SalesforceUpsert:
 * branchPercentCoverage :17.07%
 * linePercentCoverage :23.13%
 * methodPercentCoverage :57.14%
 * complexityPercentCoverage :18.75%
 * instructionPercentCoverage :23.27%
    
org.pentaho.di.trans.steps.salesforceinput.SalesforceConnection:
 * instructionPercentCoverageChange : +3.68%
 * branchPercentCoverageChange : +1.62%
 * linePercentCoverageChange : +4.19%
 * complexityPercentCoverageChange : +1.69%
 * methodPercentCoverageChange : +1.85%
    
org.pentaho.di.trans.steps.salesforceutils.SalesforceUtils:
 * branchPercentCoverage :100.00%
 * linePercentCoverage :94.74%
 * methodPercentCoverage :66.67%
 * complexityPercentCoverage :88.89%
 * instructionPercentCoverage :97.20%

**Comment by brosander** [comment]
@TatsianaKasiankova please disregard the above report, we are troubleshooting some checkstyle changes in the build

**Comment by TatsianaKasiankova** [comment]
@brosander Ok. Please merge it so that I can proceed with it.

**Comment by lucboudreau** [comment]
@TatsianaKasiankova The license header warnings are probably warranted. Please make sure that the new files include the proper headers.

**Comment by TatsianaKasiankova** [comment]
@brosander , @lucboudreau Hi,
fixed license header warnings.

File: plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceupsert/SalesforceUpsert.java [hunk 1]
```diff
@@ -39,7 +38,9 @@
 import org.pentaho.di.trans.step.StepMeta;
 import org.pentaho.di.trans.step.StepMetaInterface;
 import org.pentaho.di.trans.steps.salesforceinput.SalesforceConnection;
+import static org.pentaho.di.trans.steps.salesforceutils.SalesforceUtils.getFieldToNullName;
```
Comment (lucboudreau) [inline plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceupsert/SalesforceUpsert.java]: Please do not use static imports in runtime code.

File: plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceupdate/SalesforceUpdate.java [hunk 2]
```diff
@@ -22,6 +22,8 @@
 
 package org.pentaho.di.trans.steps.salesforceupdate;
 
+import static org.pentaho.di.trans.steps.salesforceutils.SalesforceUtils.getFieldToNullName;
```
Comment (lucboudreau) [inline plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceupdate/SalesforceUpdate.java]: Please do not use static imports in runtime code.

**Comment by TatsianaKasiankova** [comment]
@brosander, @lucboudreau 
done.

**Comment by lucboudreau** [comment]
Build Commands:
==============
Build:
------
1. rm -r ~/.kettle; rm -r ~/.pentaho; echo removed hidden folders
2. ant -Djunit.jvmargs=-XX:PermSize=256m -f plugins/salesforce/build.xml clean-all resolve jacoco jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceinsert/SalesforceInsert.java
  plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceupdate/SalesforceUpdate.java
  plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceupsert/SalesforceUpsert.java
  plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceutils/SalesforceUtils.java
  plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceutils/messages/messages_en_US.properties
  plugins/salesforce/test-src/org/pentaho/di/trans/steps/salesforceinsert/SalesforceInsertTest.java
  plugins/salesforce/test-src/org/pentaho/di/trans/steps/salesforceupdate/SalesforceUpdateTest.java
  plugins/salesforce/test-src/org/pentaho/di/trans/steps/salesforceupsert/SalesforceUpsertTest.java
  plugins/salesforce/test-src/org/pentaho/di/trans/steps/salesforceutils/SalesforceUtilsTest.java


Checkstyle violations:
======================
See: https://github.com/pentaho/pentaho-coding-standards

File plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceinsert/SalesforceInsert.java
1. Message: '<' is not followed by whitespace. at Line: 131 Column: 19
2. Message: '<' is not followed by whitespace. at Line: 131 Column: 64
3. Message: '<' is not preceded with whitespace. at Line: 131 Column: 18
4. Message: '<' is not preceded with whitespace. at Line: 131 Column: 63
5. Message: '>' is not followed by whitespace. at Line: 131 Column: 79
6. Message: '>' is not preceded with whitespace. at Line: 131 Column: 33
7. Message: '>' is not preceded with whitespace. at Line: 131 Column: 78
8. Message: '<' is not followed by whitespace. at Line: 133 Column: 19
9. Message: '<' is not followed by whitespace. at Line: 133 Column: 56
10. Message: '<' is not preceded with whitespace. at Line: 133 Column: 18
11. Message: '<' is not preceded with whitespace. at Line: 133 Column: 55
12. Message: '>' is not followed by whitespace. at Line: 133 Column: 63
13. Message: '>' is not preceded with whitespace. at Line: 133 Column: 25
14. Message: '>' is not preceded with whitespace. at Line: 133 Column: 62
15. Message: '<' is not followed by whitespace. at Line: 56 Column: 24
16. Message: '<' is not preceded with whitespace. at Line: 56 Column: 23
17. Message: '>' is not preceded with whitespace. at Line: 56 Column: 25
18. Message: '?' is not followed by whitespace. at Line: 56 Column: 25
19. Message: '?' is not preceded with whitespace. at Line: 56 Column: 24

File plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceupdate/SalesforceUpdate.java
1. Message: '<' is not followed by whitespace. at Line: 128 Column: 19
2. Message: '<' is not followed by whitespace. at Line: 128 Column: 56
3. Message: '<' is not preceded with whitespace. at Line: 128 Column: 18
4. Message: '<' is not preceded with whitespace. at Line: 128 Column: 55
5. Message: '>' is not followed by whitespace. at Line: 128 Column: 63
6. Message: '>' is not preceded with whitespace. at Line: 128 Column: 25
7. Message: '>' is not preceded with whitespace. at Line: 128 Column: 62
8. Message: '<' is not followed by whitespace. at Line: 129 Column: 19
9. Message: '<' is not followed by whitespace. at Line: 129 Column: 64
10. Message: '<' is not preceded with whitespace. at Line: 129 Column: 18
11. Message: '<' is not preceded with whitespace. at Line: 129 Column: 63
12. Message: '>' is not followed by whitespace. at Line: 129 Column: 79
13. Message: '>' is not preceded with whitespace. at Line: 129 Column: 33
14. Message: '>' is not preceded with whitespace. at Line: 129 Column: 78
15. Message: '<' is not followed by whitespace. at Line: 53 Column: 24
16. Message: '<' is not preceded with whitespace. at Line: 53 Column: 23
17. Message: '>' is not preceded with whitespace. at Line: 53 Column: 25
18. Message: '?' is not followed by whitespace. at Line: 53 Column: 25
19. Message: '?' is not preceded with whitespace. at Line: 53 Column: 24

File plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceupsert/SalesforceUpsert.java
1. Message: '<' is not followed by whitespace. at Line: 127 Column: 19
2. Message: '<' is not followed by whitespace. at Line: 127 Column: 56
3. Message: '<' is not preceded with whitespace. at Line: 127 Column: 18
4. Message: '<' is not preceded with whitespace. at Line: 127 Column: 55
5. Message: '>' is not followed by whitespace. at Line: 127 Column: 63
6. Message: '>' is not preceded with whitespace. at Line: 127 Column: 25
7. Message: '>' is not preceded with whitespace. at Line: 127 Column: 62
8. Message: '<' is not followed by whitespace. at Line: 128 Column: 19
9. Message: '<' is not followed by whitespace. at Line: 128 Column: 64
10. Message: '<' is not preceded with whitespace. at Line: 128 Column: 18
11. Message: '<' is not preceded with whitespace. at Line: 128 Column: 63
12. Message: '>' is not followed by whitespace. at Line: 128 Column: 79
13. Message: '>' is not preceded with whitespace. at Line: 128 Column: 33
14. Message: '>' is not preceded with whitespace. at Line: 128 Column: 78
15. Message: '<' is not followed by whitespace. at Line: 53 Column: 24
16. Message: '<' is not preceded with whitespace. at Line: 53 Column: 23
17. Message: '>' is not preceded with whitespace. at Line: 53 Column: 25
18. Message: '?' is not followed by whitespace. at Line: 53 Column: 25
19. Message: '?' is not preceded with whitespace. at Line: 53 Column: 24

File plugins/salesforce/src/org/pentaho/di/trans/steps/salesforceutils/SalesforceUtils.java
1. Message: '<' is not followed by whitespace. at Line: 38 Column: 24
2. Message: '<' is not preceded with whitespace. at Line: 38 Column: 23
3. Message: '>' is not preceded with whitespace. at Line: 38 Column: 25
4. Message: '?' is not followed by whitespace. at Line: 38 Column: 25
5. Message: '?' is not preceded with whitespace. at Line: 38 Column: 24

File plugins/salesforce/test-src/org/pentaho/di/trans/steps/salesforceinsert/SalesforceInsertTest.java
1. Message: '<' is not followed by whitespace. at Line: 50 Column: 26
2. Message: '<' is not preceded with whitespace. at Line: 50 Column: 25
3. Message: '>' is not preceded with whitespace. at Line: 50 Column: 68
4. Message: '<' is not followed by whitespace. at Line: 55 Column: 28
5. Message: '<' is not preceded with whitespace. at Line: 55 Column: 27
6. Message: '>' is not followed by whitespace. at Line: 55 Column: 71
7. Message: '>' is not preceded with whitespace. at Line: 55 Column: 70

File plugins/salesforce/test-src/org/pentaho/di/trans/steps/salesforceupdate/SalesforceUpdateTest.java
1. Message: '<' is not followed by whitespace. at Line: 50 Column: 26
2. Message: '<' is not preceded with whitespace. at Line: 50 Column: 25
3. Message: '>' is not preceded with whitespace. at Line: 50 Column: 68
4. Message: '<' is not followed by whitespace. at Line: 55 Column: 28
5. Message: '<' is not preceded with whitespace. at Line: 55 Column: 27
6. Message: '>' is not followed by whitespace. at Line: 55 Column: 71
7. Message: '>' is not preceded with whitespace. at Line: 55 Column: 70

File plugins/salesforce/test-src/org/pentaho/di/trans/steps/salesforceupsert/SalesforceUpsertTest.java
1. Message: '<' is not followed by whitespace. at Line: 50 Column: 26
2. Message: '<' is not preceded with whitespace. at Line: 50 Column: 25
3. Message: '>' is not preceded with whitespace. at Line: 50 Column: 68
4. Message: '<' is not followed by whitespace. at Line: 55 Column: 28
5. Message: '<' is not preceded with whitespace. at Line: 55 Column: 27
6. Message: '>' is not followed by whitespace. at Line: 55 Column: 71
7. Message: '>' is not preceded with whitespace. at Line: 55 Column: 70









Unit test coverage change:
============
org.pentaho.di.trans.steps.salesforceupdate.SalesforceUpdateMeta:
 * linePercentCoverage :0.58%
 * methodPercentCoverage :2.63%
 * complexityPercentCoverage :1.85%
 * instructionPercentCoverage :0.31%
    
org.pentaho.di.trans.steps.salesforceupsert.SalesforceUpsertMeta:
 * linePercentCoverage :0.52%
 * methodPercentCoverage :2.38%
 * complexityPercentCoverage :1.69%
 * instructionPercentCoverage :0.28%
    
org.pentaho.di.trans.steps.salesforceinsert.SalesforceInsert:
 * instructionPercentCoverageChange : +5.13%
 * branchPercentCoverageChange : +7.50%
 * linePercentCoverageChange : +2.05%
 * complexityPercentCoverageChange : +10.64%
    
org.pentaho.di.trans.steps.salesforceupdate.SalesforceUpdate:
 * branchPercentCoverage :19.44%
 * linePercentCoverage :24.43%
 * methodPercentCoverage :57.14%
 * complexityPercentCoverage :20.93%
 * instructionPercentCoverage :24.47%
    
org.pentaho.di.trans.steps.salesforceupsert.SalesforceUpsert:
 * branchPercentCoverage :17.07%
 * linePercentCoverage :23.13%
 * methodPercentCoverage :57.14%
 * complexityPercentCoverage :18.75%
 * instructionPercentCoverage :23.27%
    
org.pentaho.di.trans.steps.salesforceinput.SalesforceConnection:
 * instructionPercentCoverageChange : +3.68%
 * branchPercentCoverageChange : +1.62%
 * linePercentCoverageChange : +4.19%
 * complexityPercentCoverageChange : +1.69%
 * methodPercentCoverageChange : +1.85%
    
org.pentaho.di.trans.steps.salesforceutils.SalesforceUtils:
 * branchPercentCoverage :100.00%
 * linePercentCoverage :94.74%
 * methodPercentCoverage :66.67%
 * complexityPercentCoverage :88.89%
 * instructionPercentCoverage :97.20%

**Pull request merged**